pysb/tools/sensitivity_analysis.py

```diff
@@ -0,0 +1,627 @@
+            observable=observable,\
+            objective_function=obj_func_cell_cycle,\
+            solver=solver\
+        )
+    >>> print(sens.b_matrix)
+    [[((0.8, 'cdc0'), (0.8, 'cdc0')) ((0.8, 'cdc0'), (1.2, 'cdc0'))
+      ((0.8, 'cdc0'), (0.8, 'cyc0')) ((0.8, 'cdc0'), (1.2, 'cyc0'))]
+     [((1.2, 'cdc0'), (0.8, 'cdc0')) ((1.2, 'cdc0'), (1.2, 'cdc0'))
+      ((1.2, 'cdc0'), (0.8, 'cyc0')) ((1.2, 'cdc0'), (1.2, 'cyc0'))]
+     [((0.8, 'cyc0'), (0.8, 'cdc0')) ((0.8, 'cyc0'), (1.2, 'cdc0'))
+      ((0.8, 'cyc0'), (0.8, 'cyc0')) ((0.8, 'cyc0'), (1.2, 'cyc0'))]
+     [((1.2, 'cyc0'), (0.8, 'cdc0')) ((1.2, 'cyc0'), (1.2, 'cdc0'))
+      ((1.2, 'cyc0'), (0.8, 'cyc0')) ((1.2, 'cyc0'), (1.2, 'cyc0'))]]
+    >>> sens.run()
+    >>> print(sens.p_matrix)#doctest: +NORMALIZE_WHITESPACE
+    [[ 0.      0.      5.0243 -4.5381]
+     [ 0.      0.      5.0243 -4.5381]
+     [ 5.0243  5.0243  0.      0.    ]
+     [-4.5381 -4.5381  0.      0.    ]]
+
+    >>> print(sens.p_prime_matrix) #doctest: +NORMALIZE_WHITESPACE
+     [[ 0.      0.      5.0243 -4.5381]
+      [ 0.      0.      5.0243 -4.5381]
+      [ 0.      0.      0.      0.    ]
+      [ 0.      0.      0.      0.    ]]
+    >>> print(sens.p_matrix - sens.p_prime_matrix) \
+            #doctest: +NORMALIZE_WHITESPACE
+     [[ 0.      0.      0.      0.    ]
+      [ 0.      0.      0.      0.    ]
+      [ 5.0243  5.0243  0.      0.    ]
+      [-4.5381 -4.5381  0.      0.    ]]
+    >>> sens.create_boxplot_and_heatplot() #doctest: +ELLIPSIS
+    <matplotlib.figure.Figure object ...>
+    """
+
+    def __init__(self, solver, values_to_sample, objective_function,
+                 observable):
+        self._model = solver.model
+        self._logger = get_logger(__name__, model=self._model)
+        self._logger.info('%s created for observable %s' % (
+            self.__class__.__name__, observable))
+        generate_equations(self._model)
+        self._species_of_interest_cache = None
+        self._values_to_sample = values_to_sample
+        if solver is None or not isinstance(solver,
+                                            pysb.simulator.base.Simulator):
+            raise(TypeError, "solver must be a pysb.simulator object")
+        self._solver = solver
+        self.objective_function = objective_function
+        self.observable = observable
+
+        # Outputs
+        self.b_matrix = []
+        self.b_prime_matrix = []
+        self.p_prime_matrix = np.zeros(self._size_of_matrix)
+        self.p_matrix = np.zeros(self._size_of_matrix)
+
+        self._original_initial_conditions = np.zeros(len(self._model.species))
+        self._index_of_species_of_interest = self._create_index_of_species()
+        self.simulation_initials = self._setup_simulations()
+        # Stores the objective function value for the original unperturbed
+        # model
+        self._objective_fn_standard = None
+
+    @property
+    def _species_of_interest(self):
```

Comment:
Species is definitely more appropriate than proteins.. :)  But my point was that this appears to store **parameters** not **species**. Initial condition parameters, yes, but it's still a misleading name. Maybe `_ic_params_of_interest`?